File: version_3/code-samples/ClientApi/Commands/Documents/Put.cs
using Raven.Abstractions.Data;
using Raven.Client.Document;
using Raven.Documentation.CodeSamples.Orders;
using Raven.Json.Linq;

namespace Raven.Documentation.CodeSamples.ClientApi.Commands.Documents
{
	public class Put
    {
		private interface IFoo
		{
			 #region put_1
			 PutResult Put(string key, Etag etag, RavenJObject document, RavenJObject metadata);
			 #endregion
		}

		public Put()
		{
			using (var store = new DocumentStore())
			{
				#region put_3
				store
					.DatabaseCommands
					.Put(
						"categories/999",
						null,
						RavenJObject.FromObject(new Category
						{
							Name = "My Category",
							Description = "My Category description"
						}),
						new RavenJObject());
				#endregion
			}
		}
	}
}

File: Documentation/3.0/Samples/java/src/test/java/net/ravendb/clientapi/commands/documents/Put.java
package net.ravendb.clientapi.commands.documents;

import net.ravendb.abstractions.data.Etag;
import net.ravendb.abstractions.data.PutResult;
import net.ravendb.abstractions.json.linq.RavenJObject;
import net.ravendb.client.IDocumentStore;
import net.ravendb.client.document.DocumentStore;
import net.ravendb.samples.northwind.Category;


public class Put {
  public interface IFoo {
    //region put_1
    public PutResult put(String key, Etag guid, RavenJObject document, RavenJObject metadata);
    //endregion

  }

  public Put() throws Exception {
    try (IDocumentStore store = new DocumentStore().initialize()) {
      //region put_3
      Category category = new Category();
      category.setName("My Category");
      category.setDescription("My Category description");

      store.getDatabaseCommands().put(
        "categories/999",
        null,
        RavenJObject.fromObject(category),
        new RavenJObject());
      //endregion
    }
  }

}
